commit 1fa59d014517ac0307c3bb1e223039535bd726cd
Author: Yang Bo <yangbo@deepin.com>
Date:   Mon Jun 13 18:27:27 2016 +0800

    Fix usbkbd and usbmouse to support a narrow range of device.
    
    Signed-off-by: Yang Bo <yangbo@deepin.com>

diff --git a/drivers/hid/hid-ids.h b/drivers/hid/hid-ids.h
index 909ab01..2e69670 100644
--- a/drivers/hid/hid-ids.h
+++ b/drivers/hid/hid-ids.h
@@ -1051,4 +1051,7 @@
 #define USB_DEVICE_ID_RAPHNET_2NES2SNES	0x0002
 #define USB_DEVICE_ID_RAPHNET_4NES4SNES	0x0003
 
+#define USB_VENDOR_ID_CNTOUCH   0x294e
+#define USB_DEVICE_ID_CNTOUCH   0x1001
+
 #endif
diff --git a/drivers/hid/usbhid/usbkbd.c b/drivers/hid/usbhid/usbkbd.c
index 9a332e6..ee83a51 100644
--- a/drivers/hid/usbhid/usbkbd.c
+++ b/drivers/hid/usbhid/usbkbd.c
@@ -394,7 +394,8 @@ static void usb_kbd_disconnect(struct usb_interface *intf)
 }
 
 static struct usb_device_id usb_kbd_id_table [] = {
-	{ USB_INTERFACE_INFO(USB_INTERFACE_CLASS_HID, USB_INTERFACE_SUBCLASS_BOOT,
+	{ USB_DEVICE_AND_INTERFACE_INFO(USB_VENDOR_ID_CNTOUCH, USB_DEVICE_ID_CNTOUCH,
+        USB_INTERFACE_CLASS_HID, USB_INTERFACE_SUBCLASS_BOOT,
 		USB_INTERFACE_PROTOCOL_KEYBOARD) },
 	{ }						/* Terminating entry */
 };
diff --git a/drivers/hid/usbhid/usbmouse.c b/drivers/hid/usbhid/usbmouse.c
index bf16d72..747af6d 100644
--- a/drivers/hid/usbhid/usbmouse.c
+++ b/drivers/hid/usbhid/usbmouse.c
@@ -228,7 +228,8 @@ static void usb_mouse_disconnect(struct usb_interface *intf)
 }
 
 static struct usb_device_id usb_mouse_id_table [] = {
-	{ USB_INTERFACE_INFO(USB_INTERFACE_CLASS_HID, USB_INTERFACE_SUBCLASS_BOOT,
+	{ USB_DEVICE_AND_INTERFACE_INFO(USB_VENDOR_ID_CNTOUCH, USB_DEVICE_ID_CNTOUCH,
+        USB_INTERFACE_CLASS_HID, USB_INTERFACE_SUBCLASS_BOOT,
 		USB_INTERFACE_PROTOCOL_MOUSE) },
 	{ }	/* Terminating entry */
 };
